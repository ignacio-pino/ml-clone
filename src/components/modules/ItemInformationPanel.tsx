import { Box, Divider, Text } from "@chakra-ui/react";
import AttributeList from "../common/AttributeList";
import ImageGallery from "../common/ImageGallery";
import QuestionsList from "../common/QuestionsList";
import {
  IPictureList,
  IAttributeListData,
  IQuestionsList,
} from "../../Interfaces";

export default function ItemInformationPanel({
  pictureList,
  attributeList,
  description,
  questions,
}: {
  pictureList: IPictureList[];
  attributeList: IAttributeListData[];
  description: string;
  questions: IQuestionsList[];
}) {
  return (
    <>
      <Box pl={"3vw"} pt={"3vh"}>
        <ImageGallery pictures={pictureList} />
      </Box>
      <Box pl={"3vw"} pt="5vh">
        <Divider mb={"2rem"} backgroundColor={"gray.500"} />
        <AttributeList attributeListData={attributeList} />
        <Divider mt={"2rem"} backgroundColor={"gray.500"} />
      </Box>
      <Box pl={"3vw"} w={"40vw"} mt={"2vh"} pb={"5vh"}>
        <Text fontSize={"large"} mb={"3vh"}>
          Descripci√≥n
        </Text>
        <Text>{description}</Text>
        <Divider my={"2rem"} backgroundColor={"gray.500"} />
        <QuestionsList questionsList={questions} />
        <Divider my={"2rem"} backgroundColor={"gray.500"} />
      </Box>
    </>
  );
}
