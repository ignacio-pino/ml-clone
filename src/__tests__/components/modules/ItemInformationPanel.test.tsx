import { render, screen } from "@testing-library/react";
import ItemInformationPanel from "../../../components/modules/ItemInformationPanel";
import {
  IPictureList,
  IAttributeListData,
  IQuestionsList,
} from "../../../Interfaces";

describe("ItemInformationPanel Component", () => {
  const mockPictures: IPictureList[] = [
    { id: "1", url: "url1" },
    { id: "2", url: "url2" },
  ];

  const mockAttributes: IAttributeListData[] = [
    { id: "1", name: "Brand", value_name: "Apple" },
    { id: "2", name: "Memory", value_name: "16GB" },
  ];

  const mockQuestions: IQuestionsList[] = [
    {
      id: "1",
      status: "ANSWERED",
      text: "Is this brand new?",
      answer: {
        text: "Yes, it's brand new.",
        date_created: "2023-08-18",
      },
    },
  ];

  it("should match snapshot", () => {
    const container = render(
      <ItemInformationPanel
        pictureList={mockPictures}
        attributeList={mockAttributes}
        description="A great product."
        questions={mockQuestions}
      />
    ).container;
    expect(container).toMatchSnapshot();
  });

  it("renders image gallery correctly", () => {
    render(
      <ItemInformationPanel
        pictureList={mockPictures}
        attributeList={mockAttributes}
        description="A great product."
        questions={mockQuestions}
      />
    );
    const imageElement = screen.getByRole("img", { name: "1" });
    expect(imageElement).toBeInTheDocument();
  });

  it("renders attributes correctly", () => {
    render(
      <ItemInformationPanel
        pictureList={mockPictures}
        attributeList={mockAttributes}
        description="A great product."
        questions={mockQuestions}
      />
    );
    const attributeName = screen.getByText("Brand");
    const attributeValue = screen.getByText("Apple");

    expect(attributeName).toBeInTheDocument();
    expect(attributeValue).toBeInTheDocument();
  });

  it("renders description correctly", () => {
    render(
      <ItemInformationPanel
        pictureList={mockPictures}
        attributeList={mockAttributes}
        description="A great product."
        questions={mockQuestions}
      />
    );
    const description = screen.getByText("A great product.");
    expect(description).toBeInTheDocument();
  });

  it("renders questions correctly", () => {
    render(
      <ItemInformationPanel
        pictureList={mockPictures}
        attributeList={mockAttributes}
        description="A great product."
        questions={mockQuestions}
      />
    );
    const questionText = screen.getByText("Is this brand new?");
    const answerText = screen.getByText("Yes, it's brand new.");
    expect(questionText).toBeInTheDocument();
    expect(answerText).toBeInTheDocument();
  });
});
