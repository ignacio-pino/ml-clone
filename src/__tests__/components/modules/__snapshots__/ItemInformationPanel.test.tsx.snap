// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ItemInformationPanel Component > should match snapshot 1`] = `
<div>
  <div
    class="css-4okjp8"
  >
    <div
      class="css-k008qs"
    >
      <div
        class="chakra-stack css-jlrmky"
      >
        <button
          class="chakra-button css-bilopy"
          data-testid="imageButton-url1"
          type="button"
        >
          <img
            alt="imageButton-1"
            class="chakra-image css-7f50zt"
            src="url1"
          />
        </button>
        <button
          class="chakra-button css-bilopy"
          data-testid="imageButton-url2"
          type="button"
        >
          <img
            alt="imageButton-2"
            class="chakra-image css-7f50zt"
            src="url2"
          />
        </button>
      </div>
      <div
        class="css-wxs6zv"
      >
        <img
          alt="1"
          class="chakra-image css-zjazeo"
          src="url1"
        />
      </div>
    </div>
  </div>
  <div
    class="css-1o8rfjj"
  >
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1b5sf6j"
    />
    <div
      class="css-0"
    >
      <b
        class="chakra-text css-0"
      >
        Características
      </b>
      <div
        class="chakra-table__container css-10e72oj"
      >
        <table
          class="chakra-table css-1i60z40"
        >
          <tbody
            class="css-0"
          >
            <tr
              class="css-0"
            >
              <td
                class="css-in3yi3"
              >
                Brand
              </td>
              <td
                class="css-0"
              >
                Apple
              </td>
            </tr>
            <tr
              class="css-0"
            >
              <td
                class="css-in3yi3"
              >
                Memory
              </td>
              <td
                class="css-0"
              >
                16GB
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-pg66kc"
    />
  </div>
  <div
    class="css-1ax9nlp"
  >
    <p
      class="chakra-text css-4jn65u"
    >
      Descripción
    </p>
    <p
      class="chakra-text css-0"
    >
      A great product.
    </p>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-3icn54"
    />
    <div
      class="chakra-stack css-f5fnvy"
    >
      <p
        class="chakra-text css-0"
      >
        Preguntas y respuestas
      </p>
      <div
        class="chakra-stack css-zztjb4"
      >
        <input
          class="chakra-input css-1smfaui"
          placeholder="Escribe tu pregunta"
        />
        <button
          class="chakra-button css-1s8cab6"
          type="button"
        >
          Preguntar
        </button>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-1caknck"
        >
          <p
            class="chakra-text css-0"
          >
            Is this brand new?
          </p>
          <div
            class="chakra-stack css-1pkewcz"
          >
            <p
              class="chakra-text css-thg5cc"
            >
              L
            </p>
            <p
              class="chakra-text css-77frxq"
            >
              Yes, it's brand new.
               
              <span
                class="chakra-text css-131f7pn"
              >
                18/08/2023
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-3icn54"
    />
  </div>
</div>
`;
